syntax = "proto3";

package helloworld;

service Greeter {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc Upload(stream UploadRequest) returns (stream UploadReply) {}
}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}

message UploadRequest {
  bytes data = 1;
  // Send 0 to indicate the end of the upload
  int32 chunk_number = 2;
}

message UploadReply {
  string status = 1;
}
